Short-Answer/Algorithms_Answers.md

```diff
@@ -1 +1,22 @@
-Add your answers to the Algorithms exercises here.
\ No newline at end of file
+Add your answers to the Algorithms exercises here.
+
+Exercise I
+
+a) O(n) - I think this is linear since it starts at n^2 to get to n^3. The running time will increase with the size of the input.
```

Comment:
This is correct. Detailed explanation below.

```
a)  a = 0                      O(1)
    while (a < n * n * n):     normally O(n^3), but...
      a = a + n * n            here we add n^2 to `a` each time
```

The whole thing is similar to for loop:

```
for (a = 0; a < n * n * n; a += n * n)
```

Those `n`s effectively cancel out to:

```
for (a = 0; a < n; a += 1)     Clearly O(n)
```

Answer: `O(n)`